packages/libp2p/src/connection-manager/connection-pruner.ts

```diff
@@ -107,8 +118,8 @@ export class ConnectionPruner {
     for (const connection of sortedConnections) {
       this.log('too many connections open - closing a connection to %p', connection.remotePeer)
       // check allow list
-      const connectionInAllowList = this.allow.some((ma) => {
-        return connection.remoteAddr.toString().startsWith(ma.toString())
+      const connectionInAllowList = this.allow.some((ipNet) => {
+        return ipNet.contains(connection.remoteAddr.nodeAddress().address)
```

Comment:
Prefer `toOptions` because `nodeAddress` will throw if the protocol isn't `tcp` or `udp`:

```suggestion
        return ipNet.contains(connection.remoteAddr.toOptions().host)
```